### PR TITLE
Update dependencies (nightly-2021-09-12)

### PR DESCRIPTION
**Pull request by viper-admin**
* [x] Update rustc version to `nightly-2021-09-12`.
* [x] Manualy update outdated dependencies (see the list below).
* [x] Manualy run `cargo update`.

<details><summary>List of direct outdated dependencies:</summary>

```
$ cargo outdated --root-deps-only --workspace

analysis
================
Name        Project  Compat  Latest  Kind    Platform
----        -------  ------  ------  ----    --------
serde_json  1.0.67   ---     1.0.68  Normal  ---

vir
================
Name       Project  Compat  Latest  Kind    Platform
----       -------  ------  ------  ----    --------
syn        1.0.75   ---     1.0.76  Normal  ---
thiserror  1.0.28   ---     1.0.29  Normal  ---

vir-gen
================
Name  Project  Compat  Latest  Kind    Platform
----  -------  ------  ------  ----    --------
syn   1.0.75   ---     1.0.76  Normal  ---

prusti-specs
================
Name        Project  Compat  Latest  Kind    Platform
----        -------  ------  ------  ----    --------
serde_json  1.0.67   ---     1.0.68  Normal  ---
syn         1.0.75   ---     1.0.76  Normal  ---

prusti-server
================
Name     Project  Compat  Latest  Kind    Platform
----     -------  ------  ------  ----    --------
futures  0.1.31   ---     0.3.17  Normal  ---
reqwest  0.9.24   ---     0.11.4  Normal  ---
tokio    0.1.22   ---     1.11.0  Normal  ---
warp     0.1.23   ---     0.3.1   Normal  ---
```
</details>

@vakaras could you take care of this?

### CODE REVIEW

File: prusti-viper/src/encoder/snapshot/encoder.rs [hunk 1]
```diff
@@ -912,8 +912,8 @@ impl SnapshotEncoder {
 
                 Ok(Snapshot::Array {
                     predicate_type: predicate_type.convert_to_snapshot(),
-                    domain: self.insert_domain(domain),
-                    snap_func: self.insert_function(snap_func),
+                    _domain: self.insert_domain(domain),
```
Comment (Aurel300) [inline prusti-viper/src/encoder/snapshot/encoder.rs:915]: If these are not used at all, we can remove the fields.

**Comment by Aurel300** [comment]
It looks like you accidentally committed the empty `prusti-viper/src/encoder/salsa/mod.rs` file?

File: prusti-viper/src/encoder/snapshot/encoder.rs (hunk 1 above)
Comment (vakaras) [reply prusti-viper/src/encoder/snapshot/encoder.rs:915]: I will refactor that code later once I implement `vir-high`.

**Comment by vakaras** [comment]
> It looks like you accidentally committed the empty `prusti-viper/src/encoder/salsa/mod.rs` file?

Yes, you are right.

**Pull request merged**